File: app/src/main/java/bsu/pi_13/flowers_team/data/repository/UserRepository.java
package bsu.pi_13.flowers_team.data.repository;

import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import bsu.pi_13.flowers_team.data.db.DBHelper;

public class UserRepository {
    private final SQLiteDatabase database;

    public UserRepository(SQLiteDatabase database) {
        this.database = database;
    }

    public boolean authenticateUser(String login, String password) {
        Cursor cursor = database.query(
                DBHelper.TABLE_USERS,
                null,
                "login = ? AND password = ?",
                new String[]{login, password},
                null, null, null
        );

        boolean result = cursor.moveToFirst();
        cursor.close();
        return result;
    }

    public boolean registerUser(String login, String password) {
        if (login == null || password == null || login.isEmpty() || password.isEmpty()) return false;

        Cursor cursor = database.query(
                DBHelper.TABLE_USERS,
                null,
                "login = ?",
                new String[]{login},
                null, null, null
        );

        boolean userExists = cursor.moveToFirst();
        cursor.close();

        if (userExists) return false;

        ContentValues values = new ContentValues();
        values.put("login", login);
        values.put("password", password);
        values.put("role", "S");

        return database.insert(DBHelper.TABLE_USERS, null, values) != -1;
    }

    public String getUserLogin(int userId) {
        Cursor cursor = database.query(
                DBHelper.TABLE_USERS,
                new String[]{"login"},
                "id = ?",
                new String[]{String.valueOf(userId)},
                null, null, null
        );

        String login = "";
        if (cursor.moveToFirst()) {
            login = cursor.getString(0);
        }
        cursor.close();
        return login;
    }

    public int getUserIdByLogin(String login) {
        Cursor cursor = database.query(
                DBHelper.TABLE_USERS,
                new String[]{"id"},
                "login = ?",
                new String[]{login},
                null, null, null
        );

        int userId = -1;
        if (cursor.moveToFirst()) {
            userId = cursor.getInt(0);
        }
        cursor.close();
        return userId;
    }
}

File: app/src/main/java/bsu/pi_13/flowers_team/data/repository/UserRepository.kt
package bsu.pi_13.flowers_team.data.repository
import android.content.ContentValues
import android.database.sqlite.SQLiteDatabase
import bsu.pi_13.flowers_team.data.db.DBHelper
import bsu.pi_13.flowers_team.data.db.DBHelper.TABLE_USERS


class UserRepository(private val database: SQLiteDatabase) {
    fun authenticateUser(login: String, password: String): Boolean {
        val cursor = database.query(
            DBHelper.TABLE_USERS,
            null,
            "login = ? AND password = ?",
            arrayOf(login, password),
            null, null, null
        )
        val result = cursor.count > 0
        cursor.close()
        return result
    }

    fun registerUser(login: String?, password: String?): Boolean {
        if (login == null || password == null || login.isEmpty() || password.isEmpty()) {
            return false
        }

        val cursor = database.query(
            DBHelper.TABLE_USERS,
            null,
            "login = ?",
            arrayOf(login),
            null, null, null
        )
        val userExists = cursor.count > 0
        cursor.close()

        if (userExists) {
            return false
        }

        val values = ContentValues()
        values.put("login", login)
        values.put("password", password)
        values.put("role", "S")

        return database.insert(DBHelper.TABLE_USERS, null, values) != -1L
    }

    fun getUserIdByLogin(login: String): Int {
        val cursor = database.query(
            TABLE_USERS,
            arrayOf("id"),
            "login = ?",
            arrayOf(login),
            null, null, null
        )
        return if (cursor.moveToFirst()) {
            cursor.getInt(0)
        } else {
            -1
        }.also { cursor.close() }
    }
}
